gateway/webhook/webhook_test.go

```diff
@@ -518,3 +518,96 @@ func TestPrepareRequestBody(t *testing.T) {
 		})
 	}
 }
+
+func TestAllowGetReqForWebhookSrc(t *testing.T) {
+	initWebhook()
+	ctrl := gomock.NewController(t)
+	mockGW := mockWebhook.NewMockGateway(ctrl)
+	transformerServer := httptest.NewServer(
+		http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			defer func() { _ = r.Body.Close() }()
+			body, _ := io.ReadAll(r.Body)
+			var requests []interface{}
+			_ = json.Unmarshal(body, &requests)
+			var responses []transformerResponse
+			for i := 0; i < len(requests); i++ {
+				responses = append(responses, transformerResponse{
+					OutputToSource: outputToWebhook,
+					StatusCode:     http.StatusOK,
+				})
+			}
+			respBody, _ := json.Marshal(responses)
+			_, _ = w.Write(respBody)
+		}))
+	wbh := Setup(mockGW, transformer.NewNoOpService(), stats.Default, func(bt *batchWebhookTransformerT) {
+		bt.sourceTransformAdapter = getMockSourceTransformAdapterFunc(transformerServer.URL)
+	})
+	type testCase struct {
+		description string
+		input       struct {
+			srcMap map[string]struct{}
+			method string
+			srcDef string
+		}
+		output bool
+	}
+
+	cases := []testCase{
```

Comment:
[optional]: We can use an anonymous struct in here.
```suggestion
	[]struct {
		description string
		input       struct {
			srcMap map[string]struct{}
			method string
			srcDef string
		}
		output bool
	}{
```